src/alcotest.ml

```diff
@@ -617,7 +619,12 @@ let int64 = testable Fmt.int64 (=)
 
 let int = testable Fmt.int (=)
 
-let float eps = testable Fmt.float (fun x y -> abs_float (x -. y) <= eps)
+let float eps =
+  let same x y = (isnan x && isnan y) ||
```

Comment:
what do you think about naming each case to help reading operators? Something like:
```ocaml
let both_are_nan = isnan x && isnan y in
...
both_are_nan || ...
```

Reply:
Does it bring that much clarity? I'm not convinced

Reply:
I think that it would help with understanding operator priority and add an opportunity to explain why the `x = y` case is important.

Reply:
Ideally I would use `Float.equal x y || abs_float (x -. y) <= eps`.

In any case I'm not sure how would using `both_are_nan` would clarify the second term is related to infinity values, will a comment do?

Reply:
Sure, `Float.equal` works for me!

Reply:
Sorry, I forgot to say why that's the ideal solution: it's not available on Ocaml < 4.07. (4.04 to 4.06 seem to be supported)